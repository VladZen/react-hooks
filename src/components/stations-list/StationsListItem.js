import StationType from '@/types/Station'

import Status from '@/components/stations-list/StationStatus.js'
import { Link } from 'react-router-dom'

const StationsListItem = ({ station }) => {
  const {
    available,
    name,
    station_ID
  } = station

  return (
    <li className='stations__item'>
      <Link
        className='stations__item__name text-ellipsis'
        title={ name }
        to={ `/stations/${station_ID}` }
      >
        { name }
      </Link>

      <Status available={ available } />
    </li>
  )
}

StationsListItem.propTypes = {
  station: StationType
}

export default StationsListItem
