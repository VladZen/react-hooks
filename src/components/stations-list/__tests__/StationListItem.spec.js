import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import StationsListItem from '@/components/stations-list/StationsListItem'

jest.mock('@/components/stations-list/StationStatus', () => {
  return function Mock (props) {
    return (
      <div data-testid="list-item">
        { props.available ? 'available' : 'offline' }
      </div>
    )
  }
})

const mockStations = {
  available: {
    'station_ID': 102,
    'custom_evse_id': null,
    'location_ID': 102,
    'seller_ID': 27,
    'name': 'Endesa 2',
    'connected': 1,
    'position': '60.868623,26.702901',
    'available': 1,
    'lastconnect': '2019-02-22T13:36:12.000Z',
    'roaming_identifier_cpo': 'FI*001'
  },
  offline: {
    'station_ID': 103,
    'custom_evse_id': null,
    'location_ID': 103,
    'seller_ID': 28,
    'name': 'Endesa 2',
    'connected': 1,
    'position': '60.868623,26.702901',
    'available': 0,
    'lastconnect': '2019-02-22T13:36:12.000Z',
    'roaming_identifier_cpo': 'FI*001'
  }
}

describe('StationListItem component', () => {
  describe('rendering', () => {
    it.each(['available', 'offline'])('%s', (status) => {
      const { container } = render(
        <MemoryRouter>
          <StationsListItem station={ mockStations[status] } />
        </MemoryRouter>
      )

      expect(container).toMatchSnapshot()
    })
  })
})
