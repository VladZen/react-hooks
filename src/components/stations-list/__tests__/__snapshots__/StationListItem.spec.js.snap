// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationListItem component rendering available 1`] = `
<div>
  <li
    class="stations__item"
  >
    <a
      class="stations__item__name text-ellipsis"
      href="/stations/102"
      title="Endesa 2"
    >
      Endesa 2
    </a>
    <div
      data-testid="list-item"
    >
      available
    </div>
  </li>
</div>
`;

exports[`StationListItem component rendering offline 1`] = `
<div>
  <li
    class="stations__item"
  >
    <a
      class="stations__item__name text-ellipsis"
      href="/stations/103"
      title="Endesa 2"
    >
      Endesa 2
    </a>
    <div
      data-testid="list-item"
    >
      offline
    </div>
  </li>
</div>
`;
